### PR TITLE
Try a direct connection only if there isn't one already

### PR DESCRIPTION
**Pull request by diegomrsantos**

### CODE REVIEW

File: libp2p/services/hpservice.nim [hunk 1]
```diff
@@ -71,22 +96,10 @@ method setup*(self: HPService, switch: Switch): Future[bool] {.async.} =
...
-            natAddrs =  switch.peerInfo.listenAddrs.mapIt(switch.peerStore.guessDialableAddr(it))
-          await dcutrClient.startSync(switch, peerId, natAddrs)
-          await sleepAsync(2000.milliseconds) # grace period before closing relayed connection
-          await conn.close()
-      except CatchableError as err:
-        debug "Hole punching failed during dcutr", err = err.msg
+    proc newConnectedPeerHandler(peerId: PeerId, event: PeerEvent) {.async.} =
+      await newConnectedPeerHandlerAux(self, switch, peerId, event)
+
+    self.newConnectedPeerHandler = newConnectedPeerHandler
```
Comment (Menduist) [inline libp2p/services/hpservice.nim:102]:
```suggestion
    self.newConnectedPeerHandler = proc (peerId: PeerId, event: PeerEvent) {.async.} =
      await newConnectedPeerHandlerAux(self, switch, peerId, event)
```
And you cane rename newConnectedPeerHandlerAux to onNewPeer or something like that

File: libp2p/services/hpservice.nim [hunk 2]
```diff
@@ -63,6 +63,31 @@ proc tryStartingDirectConn(self: HPService, switch: Switch, peerId: PeerId): Fut
...
+    # Get all connections to the peer. If there is at least one non-relayed connection, return.
+    var relayedConnt: Connection
+    for muxer in switch.connManager.getConnections()[peerId]:
+      let conn = muxer.connection
+      if not isRelayed(conn):
+        return
+      elif conn.transportDir == Direction.In:
+        relayedConnt = conn
+
+    if not isNil(relayedConnt):
```
Comment (Menduist) [inline libp2p/services/hpservice.nim:77]:
```suggestion
    let connections = switch.connManager.getConnections()[peerId].mapIt(it.connection)
    if connections.anyIt(not isRelayed(it)):
      return
    let incomingRelays = connections.filterIt(it.transportDir == Direction.In)
    if incomingRelays.len == 0:
      return
    
    let relayedConn = incomingRelays[0]
```
(need to de-ident the rest of the code)
wdyt?

**Review by Menduist — APPROVED** [review]
LGTM besides style questions

File: libp2p/services/hpservice.nim (hunk 2 above)
Comment (diegomrsantos) [reply libp2p/services/hpservice.nim:77]: Not sure if this is better

Comment (Menduist) [reply libp2p/services/hpservice.nim]: I think the current version is slightly confusing, I tried to convey the intention more clearly

Comment (diegomrsantos) [reply libp2p/services/hpservice.nim]: done

**Review by Menduist — APPROVED** [review]

**Pull request merged**